### ISSUE TITLE
Unhandled JS exception: Can't find variable require

### ISSUE THREAD

**Issue by zanisis** [expo/expo-cli#276]
# Required format for issues

i cannot running ios emulator and android emulator

## Description

still error message can't find variable AppEntry.bundle?.

i don't know what should i do my package json just redux redux-thunk and babel-preset-expo but still not running 2 device emulator android or ios

## Expected Behavior

What action did you perform, and what did you expect to happen?

## Observed Behavior

What actually happened when you performed the above actions?

If there's an error message, please paste the _full terminal output and error message_ in this code block:

[screen shot 2018-12-28 at 15 16 06](https://user-images.githubusercontent.com/25142795/50508419-6e27bd80-0ab4-11e9-8631-a1e1ab03b808.png)

## Environment

  Expo CLI 2.6.14 environment info:
    System:
      OS: macOS High Sierra 10.13.6
      Shell: 5.3 - /bin/zsh
    Binaries:
      Node: 10.15.0 - /usr/local/bin/node
      Yarn: 1.12.3 - /usr/local/bin/yarn
      npm: 6.4.1 - /usr/local/bin/npm
      Watchman: 4.9.0 - /usr/local/bin/watchman
    IDEs:
      Android Studio: 3.1 AI-173.4907809
      Xcode: 10.1/10B61 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^31.0.2 => 31.0.6
      react: 16.5.0 => 16.5.0
      react-native: https://github.com/expo/react-native/archive/sdk-31.0.0.tar.gz => 0.57.1
      react-navigation: ^3.0.9 => 3.0.9
    npmGlobalPackages:
      expo-cli: 2.6.14

Also specify:

ios emulator device iPhone 8
android emulator nexus_5X

## Reproducible Demo

Please provide a minimized reproducible demonstration of the problem you're reporting.

Issues that come with minimal repro's are resolved much more quickly than issues where a maintainer has to reproduce themselves.

**Comment by fson** [comment]
Could you try clearing the transform cache of Metro bundler by starting the project with `expo start --clear`?

**Comment by zanisis** [comment]
> Could you try clearing the transform cache of Metro bundler by starting the project with `expo start --clear`?

already running with expo start --clear but still can't find variable: require.
i try with build new project using expo init and try copy all file src, babel, package.json from old project (project error) after running expo start still running not error i don't know what happen?.

**Comment by fson** [comment]
Alright, sounds like the problem got solved by itself and it's not really reproducible.

If you still have isssues, please re-open the issue.

**Comment by tiagob** [comment]
Same. Was working fine a couple weeks ago and ran into this error today (no code changes). I had setup Typescript with devDependencies from
https://blogs.msdn.microsoft.com/typescript/2018/08/27/typescript-and-babel-7/

```json
  "devDependencies": {
    "@babel/plugin-proposal-class-properties": "^7.2.3",
    "@babel/plugin-proposal-object-rest-spread": "^7.2.0",
    "@babel/preset-env": "^7.2.3",
    "@babel/preset-typescript": "^7.1.0",
```

and their `.babelrc`

I removed these and everything worked. Confirmed it's using my custom `tsconfig.json`

@fson would be great to see TypeScript with expo documentation. A bunch of low-quality articles online pre v31. Users still want to `yarn add @types/expo @types/react @types/react-native -D` and most likely add a custom `tsconfig.json`

**Comment by tiagob** [comment]
Could be something with the new app version released Jan 4 https://itunes.apple.com/us/app/expo-client/id982107779?mt=8

**Comment by alexborton** [comment]
Just to add to this. I came across the same issue trying to port an Expo build into [react-boilerplate](https://github.com/react-boilerplate/react-boilerplate) (to start a hybrid application).

[react-boilerplate](https://github.com/react-boilerplate/react-boilerplate) has a number of presets in the `babel.config.js`. One is `@babel/preset-env` as mentioned above. It seems to be this that is the culprit. 

I don't know a great deal about `@babel/preset-env` and it might well be that there is a configuration that would allow Expo to work properly. If anyone has some more knowledge on that, that would be great :)

**Comment by MuFa117** [comment]
Thanks @alexborton , your post moved me towards some solutions (at least temporary one). 
I've been porting my app from SDK25 to SDK32 and I have created `babel.config.js` looking like this:
`module.exports = function(api) {
  api.cache(true);
  return {
    presets: ['babel-preset-expo'],
  };
};`
And everything was working just fine. Then I started adding all web stuff - including whole @babel libs, so I ended up adding `.babelrc` file to my project as well looking like this:
`{
  "presets": ["@babel/preset-env", "@babel/preset-react"],
  "plugins": ["@babel/plugin-proposal-class-properties", "@babel/plugin-proposal-object-rest-spread"]
}
`

As of now, I ended up of just removing/renaming `.babelrc` file, as adding `babel-preset-expo` to it didn't work out, and my project is rushing. However, that helped me out and I was able to start out the project. Might get back to it later on, to figured out the way to have both of them running, but figured out that I'd share my findings.
